src/distributed_scheduler.rs

```diff
@@ -769,17 +767,14 @@ impl DistributedScheduler {
         Vec::new()
     }
 
-    fn wait_for_event(&mut self, run_id: usize, timeout: i64) -> Option<CompletionEvent> {
-        let timer = SystemTime::now();
-        let end_time = timer.elapsed().unwrap().as_millis() + timeout as u128;
+    fn wait_for_event(&mut self, run_id: usize, timeout: u64) -> Option<CompletionEvent> {
+        let end = Instant::now() + Duration::from_millis(timeout);
         while self.event_queues.lock().get(&run_id).unwrap().is_empty() {
-            let time = timer.elapsed().unwrap().as_millis();
-            if time >= end_time {
+            if Instant::now() > end {
                 return None;
-            } else {
-                let dur = time::Duration::from_millis((end_time - time) as u64);
-                thread::sleep(dur);
             }
+
+            thread::sleep(Duration::from_millis(250));
```

Comment:
I've changed this to 250ms, as the previous value which was the entire duration which I don't think was intended. The same goes for local_scheduler.rs